src/crypto/aes.cpp

```diff
@@ -429,7 +429,7 @@ openssl_thread_config::openssl_thread_config()
 }
 openssl_thread_config::~openssl_thread_config()
 {
-  if (CRYPTO_get_id_callback() == &get_thread_id)
+  if (CRYPTO_get_id_callback() != NULL && CRYPTO_get_id_callback() == &get_thread_id)
```

Comment:
Isn't the first check redundant? ``&get_thread_id``will never be NULL.

Reply:
Yes, it is redundant. It did avoid a warning, but I would prefer to not add the logic. I took it back out, and the warning returned. But I think it better to live with the warning.

Reply:
What does the warning say? Are we missing something?

Reply:
Yes, including what I am talking about in the note would have been smart. Adding the ``CRYPTO_get_id_callback() != nullptr &&`` to the ``if`` fixed the warning, but I hate adding such code just to squelch a warning. 

This is the original warning:
```/src/crypto/aes.cpp: In destructor ‘fc::openssl_thread_config::~openssl_thread_config()’:
/src/crypto/aes.cpp:432:32: warning: the address of ‘static long unsigned int fc::openssl_thread_config::get_thread_id()’ will never be NULL [-Waddress]
   if (CRYPTO_get_id_callback() == &get_thread_id)

Reply:
``CRYPTO_get_id_callback`` is deprecated - does the warning persist if you switch to ``CRYPTO_THREADID_[gs]et_callback``? https://linux.die.net/man/3/crypto_lock